### ISSUE TITLE
Custom Script not running "On Download"

### ISSUE THREAD

**Issue by ajgillis04** [Radarr/Radarr#2519]
**Description:** I am trying to configure Custom Script [MP4 Automator](https://github.com/mdhiggins/sickbeard_mp4_automator/blob/master/README.md) and I can not get it to run. I changed the options to add "On Grab" and than the script executes. Searching the logs CustomScript is only found "On Grab" but not for "On Download". I have not configured a custom script in radarr before so i maybe doing something wrong. I will say the same command is working in sickrage though. Any help would be appreciated.

**Custom Script config:** https://snag.gy/KYwin1.jpg

**Radarr Version:** 0.2.0.935 Develop
**OS:** Windows 10

**Debug Logs:** 
18-2-13 18:33:03.6|Debug|NotificationService|No tags set for this notification.
18-2-13 18:33:03.6|Debug|CustomScript|Executing external script: D:\\Scripts\\Python27\\python.exe
18-2-13 18:33:03.7|Debug|python.exe|Starting D:\\Scripts\\Python27\\python.exe D:\\Scripts\\mp4_automator\\postRadarr.py >d:\\Scripts\\logs\\postradarrun.txt
18-2-13 18:33:06.3|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:33:06.3|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:33:06.3|Debug|Parser|Language parsed: English
18-2-13 18:33:06.3|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:33:06.3|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:33:06.3|Debug|Parser|Release Group parsed:
18-2-13 18:33:06.4|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (4 ms)
18-2-13 18:33:14.8|Debug|python.exe|2018-02-13 18:33:14 - RadarrPostConversion - INFO - Radarr extra script post processing started.
18-2-13 18:33:14.8|Debug|python.exe|2018-02-13 18:33:14 - readSettings - INFO - D:\\Scripts\\Python27\\python.exe
18-2-13 18:33:14.8|Debug|python.exe|2018-02-13 18:33:14 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
18-2-13 18:33:16.6|Debug|python.exe|2018-02-13 18:33:16 - RadarrPostConversion - INFO - {\'TMP\': \'C:\\\\Windows\\\\TEMP\', \'COMPUTERNAME\': \'PCName\', \'USERDOMAIN\': \'Domain\', \'PSMODULEPATH\': \'C:\\\\Program Files\\\\WindowsPowerShell\\\\Modules;C:\\\\Windows\\\\system32\\\\WindowsPowerShell\\\\v1.0\\\\Modules\', \'COMMONPROGRAMFILES\': \'C:\\\\Program Files (x86)\\\\Common Files\', \'PROCESSOR_IDENTIFIER\': \'Intel64 Family 6 Model 58 Stepping 9, GenuineIntel\', \'RADARR_RELEASE_SIZE\': \'734317760\', \'PROGRAMFILES\': \'C:\\\\Program Files (x86)\', \'PROCESSOR_REVISION\': \'3a09\', \'SYSTEMROOT\': \'C:\\\\Windows\', \'PATH\': \'C:\\\\ProgramData\\\\Oracle\\\\Java\\\\javapath;C:\\\\Windows\\\\system32;C:\\\\Windows;C:\\\\Windows\\\\System32\\\\Wbem;C:\\\\Windows\\\\System32\\\\WindowsPowerShell\\\\v1.0\\\\;C:\\\\Program Files (x86)\\\\Skype\\\\Phone\\\\;D:\\\\Scripts\\\\Python27;C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Local\\\\Microsoft\\\\WindowsApps\', \'PROGRAMFILES(X86)\': \'C:\\\\Program Files (x86)\', \'TEMP\': \'C:\\\\Windows\\\\TEMP\', \'RADARR_EVENTTYPE\': \'Grab\', \'COMMONPROGRAMFILES(X86)\': \'C:\\\\Program Files (x86)\\\\Common Files\', \'PROCESSOR_ARCHITECTURE\': \'x86\', \'RADARR_MOVIE_TMDBID\': \'605\', \'ALLUSERSPROFILE\': \'C:\\\\ProgramData\', \'LOCALAPPDATA\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Local\', \'RADARR_RELEASE_RELEASEGROUP\': \'\', \'PROGRAMW6432\': \'C:\\\\Program Files\', \'USERNAME\': \'PCName$\', \'RADARR_RELEASE_INDEXER\': \'Jacket-BT-Scene\', \'COMSPEC\': \'C:\\\\Windows\\\\system32\\\\cmd.exe\', \'RADARR_MOVIE_ID\': \'741\', \'PROGRAMDATA\': \'C:\\\\ProgramData\', \'RADARR_RELEASE_QUALITY\': \'Bluray-720p\', \'PATHEXT\': \'.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC\', \'ASL.LOG\': \'Destination=file\', \'WINDIR\': \'C:\\\\Windows\', \'APPDATA\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Roaming\', \'RADARR_RELEASE_QUALITYVERSION\': \'1\', \'SYSTEMDRIVE\': \'C:\', \'RADARR_RELEASE_TITLE\': \'Test - (2003) 720p BrRip x264 - 700mb - YIFY\', \'RADARR_MOVIE_TITLE\': \'Test\', \'NUMBER_OF_PROCESSORS\': \'8\', \'PROCESSOR_LEVEL\': \'6\', \'PROCESSOR_ARCHITEW6432\': \'AMD64\', \'COMMONPROGRAMW6432\': \'C:\\\\Program Files\\\\Common Files\', \'OS\': \'Windows_NT\', \'PUBLIC\': \'C:\\\\Users\\\\Public\', \'RADARR_MOVIE_IMDBID\': \'tt000000000\', \'USERPROFILE\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\'}
18-2-13 18:33:16.6|Debug|CustomScript|Executed external script: D:\\Scripts\\Python27\\python.exe - Status: 0
18-2-13 18:33:16.6|Debug|CustomScript|Script Output:
2/13/2018 11:33:14 PM - Standard - 2018-02-13 18:33:14 - RadarrPostConversion - INFO - Radarr extra script post processing started.
2/13/2018 11:33:14 PM - Standard - 2018-02-13 18:33:14 - readSettings - INFO - D:\\Scripts\\Python27\\python.exe
2/13/2018 11:33:14 PM - Standard - 2018-02-13 18:33:14 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
2/13/2018 11:33:16 PM - Standard - 2018-02-13 18:33:16 - RadarrPostConversion - INFO - {\'TMP\': \'C:\\\\Windows\\\\TEMP\', \'COMPUTERNAME\': \'PCName\', \'USERDOMAIN\': \'Domain\', \'PSMODULEPATH\': \'C:\\\\Program Files\\\\WindowsPowerShell\\\\Modules;C:\\\\Windows\\\\system32\\\\WindowsPowerShell\\\\v1.0\\\\Modules\', \'COMMONPROGRAMFILES\': \'C:\\\\Program Files (x86)\\\\Common Files\', \'PROCESSOR_IDENTIFIER\': \'Intel64 Family 6 Model 58 Stepping 9, GenuineIntel\', \'RADARR_RELEASE_SIZE\': \'734317760\', \'PROGRAMFILES\': \'C:\\\\Program Files (x86)\', \'PROCESSOR_REVISION\': \'3a09\', \'SYSTEMROOT\': \'C:\\\\Windows\', \'PATH\': \'C:\\\\ProgramData\\\\Oracle\\\\Java\\\\javapath;C:\\\\Windows\\\\system32;C:\\\\Windows;C:\\\\Windows\\\\System32\\\\Wbem;C:\\\\Windows\\\\System32\\\\WindowsPowerShell\\\\v1.0\\\\;C:\\\\Program Files (x86)\\\\Skype\\\\Phone\\\\;D:\\\\Scripts\\\\Python27;C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Local\\\\Microsoft\\\\WindowsApps\', \'PROGRAMFILES(X86)\': \'C:\\\\Program Files (x86)\', \'TEMP\': \'C:\\\\Windows\\\\TEMP\', \'RADARR_EVENTTYPE\': \'Grab\', \'COMMONPROGRAMFILES(X86)\': \'C:\\\\Program Files (x86)\\\\Common Files\', \'PROCESSOR_ARCHITECTURE\': \'x86\', \'RADARR_MOVIE_TMDBID\': \'605\', \'ALLUSERSPROFILE\': \'C:\\\\ProgramData\', \'LOCALAPPDATA\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Local\', \'RADARR_RELEASE_RELEASEGROUP\': \'\', \'PROGRAMW6432\': \'C:\\\\Program Files\', \'USERNAME\': \'PCName$\', \'RADARR_RELEASE_INDEXER\': \'Jacket-BT-Scene\', \'COMSPEC\': \'C:\\\\Windows\\\\system32\\\\cmd.exe\', \'RADARR_MOVIE_ID\': \'741\', \'PROGRAMDATA\': \'C:\\\\ProgramData\', \'RADARR_RELEASE_QUALITY\': \'Bluray-720p\', \'PATHEXT\': \'.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC\', \'ASL.LOG\': \'Destination=file\', \'WINDIR\': \'C:\\\\Windows\', \'APPDATA\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\AppData\\\\Roaming\', \'RADARR_RELEASE_QUALITYVERSION\': \'1\', \'SYSTEMDRIVE\': \'C:\', \'RADARR_RELEASE_TITLE\': \'Test - (2003) 720p BrRip x264 - 700mb - YIFY\', \'RADARR_MOVIE_TITLE\': \'Test\', \'NUMBER_OF_PROCESSORS\': \'8\', \'PROCESSOR_LEVEL\': \'6\', \'PROCESSOR_ARCHITEW6432\': \'AMD64\', \'COMMONPROGRAMW6432\': \'C:\\\\Program Files\\\\Common Files\', \'OS\': \'Windows_NT\', \'PUBLIC\': \'C:\\\\Users\\\\Public\', \'RADARR_MOVIE_IMDBID\': \'tt000000000\', \'USERPROFILE\': \'C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\'}
18-2-13 18:33:16.6|Debug|Api|[POST] /api/release: 200.OK (13929 ms)
18-2-13 18:33:37.5|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:33:37.5|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:33:37.5|Debug|Parser|Language parsed: English
18-2-13 18:33:37.5|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:33:37.5|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:33:37.5|Debug|Parser|Release Group parsed:
18-2-13 18:33:37.6|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (7 ms)
18-2-13 18:34:06.4|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:34:06.4|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:34:06.4|Debug|Parser|Language parsed: English
18-2-13 18:34:06.4|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:34:06.4|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:34:06.4|Debug|Parser|Release Group parsed:
18-2-13 18:34:06.4|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (7 ms)
18-2-13 18:34:36.4|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:34:36.4|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:34:36.4|Debug|Parser|Language parsed: English
18-2-13 18:34:36.4|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:34:36.4|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:34:36.4|Debug|Parser|Release Group parsed:
18-2-13 18:34:36.5|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (4 ms)
18-2-13 18:35:06.4|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:35:06.4|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:35:06.4|Debug|Parser|Language parsed: English
18-2-13 18:35:06.4|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:35:06.4|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:35:06.4|Debug|Parser|Release Group parsed:
18-2-13 18:35:06.5|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (6 ms)
18-2-13 18:35:37.6|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:35:37.6|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:35:37.6|Debug|Parser|Language parsed: English
18-2-13 18:35:37.6|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:35:37.6|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:35:37.6|Debug|Parser|Release Group parsed:
18-2-13 18:35:37.7|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (6 ms)
18-2-13 18:36:06.4|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:36:06.4|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:36:06.4|Debug|Parser|Language parsed: English
18-2-13 18:36:06.4|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:36:06.4|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:36:06.5|Debug|Parser|Release Group parsed:
18-2-13 18:36:06.5|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (4 ms)
18-2-13 18:36:36.5|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:36:36.5|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:36:36.5|Debug|Parser|Language parsed: English
18-2-13 18:36:36.5|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:36:36.5|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:36:36.5|Debug|Parser|Release Group parsed:
18-2-13 18:36:36.5|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (7 ms)
18-2-13 18:37:06.5|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:37:06.5|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:37:06.5|Debug|Parser|Language parsed: English
18-2-13 18:37:06.5|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:37:06.5|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:37:06.5|Debug|Parser|Release Group parsed:
18-2-13 18:37:06.5|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (6 ms)
18-2-13 18:37:37.8|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:37:37.8|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:37:37.8|Debug|Parser|Language parsed: English
18-2-13 18:37:37.8|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:37:37.8|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:37:37.8|Debug|Parser|Release Group parsed:
18-2-13 18:37:37.9|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (6 ms)
18-2-13 18:38:06.5|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:38:06.5|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:38:06.5|Debug|Parser|Language parsed: English
18-2-13 18:38:06.5|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:38:06.5|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:38:06.5|Debug|Parser|Release Group parsed:
18-2-13 18:38:06.6|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (7 ms)
18-2-13 18:38:36.5|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:38:36.6|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:38:36.6|Debug|Parser|Language parsed: English
18-2-13 18:38:36.6|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:38:36.6|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:38:36.6|Debug|Parser|Release Group parsed:
18-2-13 18:38:36.6|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (11 ms)
18-2-13 18:39:06.6|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:39:06.6|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:39:06.6|Debug|Parser|Language parsed: English
18-2-13 18:39:06.6|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:39:06.6|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:39:06.6|Debug|Parser|Release Group parsed:
18-2-13 18:39:06.6|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (5 ms)
18-2-13 18:39:36.6|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:39:36.6|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:39:36.6|Debug|Parser|Language parsed: English
18-2-13 18:39:36.6|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:39:36.6|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:39:36.6|Debug|Parser|Release Group parsed:
18-2-13 18:39:36.7|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (5 ms)
18-2-13 18:40:07.8|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:40:07.8|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:40:07.8|Debug|Parser|Language parsed: English
18-2-13 18:40:07.8|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:40:07.8|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:40:07.8|Debug|Parser|Release Group parsed:
18-2-13 18:40:07.9|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (6 ms)
18-2-13 18:40:36.7|Debug|Parser|Parsing string \'Test - (2003)\'
18-2-13 18:40:36.7|Debug|Parser|Movie Parsed. Test - 2003
18-2-13 18:40:36.7|Debug|Parser|Language parsed: English
18-2-13 18:40:36.7|Debug|QualityParser|Trying to parse quality for Test - (2003)
18-2-13 18:40:36.7|Debug|Parser|Quality parsed: Unknown v1
18-2-13 18:40:36.7|Debug|Parser|Release Group parsed:
18-2-13 18:40:36.7|Debug|Api|[GET] /api/queue?sort_by=timeleft&order=asc: 200.OK (4 ms)

**Comment by galli-leo** [comment]
Do you have any debug logs for the time the download finished?

**Comment by ajgillis04** [comment]
I believe the original snippet had everything up to and including a download complete. Regardless I have attached a full two hours of the debug log which should include everything. I hope that helps and thank you.

[radarrdebug.txt](https://github.com/Radarr/Radarr/files/1725868/radarrdebug.txt)

**Comment by galli-leo** [comment]
So I don't see anything in the debug log that suggests Radarr imported a file and thus would execute the script. Did you check in your history whether Radarr imported a file after setting up the on download script?

**Comment by ajgillis04** [comment]
Your response clued me in that no import was occurring. I figured out with "Completed Download Handling" turned off the movie was never getting imported and the "On Downloaded" never triggered. I had disabled it based on MP4 Automators readme instructions. From the readme "Set the Drone Factory Interval' to 0 to disable it, and disable 'Completed Download Handling' in Radarr settings. The script will trigger a specific path re-scan, allowing the mp4 conversion to be completed before Radarr starts moving stuff around."
So it appears this is not a Radarr problem. It is either a configuration problem (I've done something wrong), or a problem with the MP4 Automator scripts.
I'll keep digging thanks for you help
